pages/storage-page.ts

```diff
@@ -1,33 +1,58 @@
-import { expect, Locator, Page } from '@playwright/test';
+import { expect, type Locator, type Page } from '@playwright/test';
 
 export class StoragePage {
     readonly page: Page;
     readonly useEncryptionToggle: Locator;
     readonly encryptionEnabled: Locator;
     readonly useLVMToggle: Locator;
     readonly backButton: Locator;
+    readonly actionsButton: Locator;
+    readonly addFileSystemButton: Locator;
+    readonly resetToDefaultsButton: Locator;
+    readonly acceptButton: Locator;
+    readonly swapPartitionText: Locator;
 
     constructor(page: Page) {
-        this.page = page;
-        this.useEncryptionToggle = page.getByRole('checkbox', { name: 'Use encryption' });
-        this.encryptionEnabled = page.getByRole('button', { name: 'Encryption settings' });
-        this.useLVMToggle = page.locator('label').filter({ hasText: /Use logical volume management/ }).locator('span').first();
-        this.backButton = page.getByRole('button', { name: 'Back' });
+      this.page = page;
+      this.useEncryptionToggle = page.getByRole('checkbox', { name: 'Use encryption' });
+      this.encryptionEnabled = page.getByRole('button', { name: 'Encryption settings' });
+      this.useLVMToggle = page.locator('label').filter({ hasText: /Use logical volume management/ }).locator('span').first();
+      this.backButton = page.getByRole('button', { name: 'Back' });
+      this.actionsButton = page.getByRole('button', { name: 'Actions' }).first();
+      this.addFileSystemButton = page.getByText('Add file system', { exact: true });
+      this.resetToDefaultsButton = page.getByText('Reset to defaults', { exact: true });
+      this.acceptButton = page.getByRole('button', { name: 'Accept' });
+      this.swapPartitionText = page.getByText('Swap partition');
     }
 
     async useEncryption() {
-        await this.useEncryptionToggle.check();
+      await this.useEncryptionToggle.check();
     }
-    
+  
     async validateEncryptionIsUsed() {
-        expect(await this.useEncryptionToggle.isChecked()).toBeTruthy();
+      expect(await this.useEncryptionToggle.isChecked()).toBeTruthy();
     }
-    
+
     async useLVM() {
-        await this.useLVMToggle.click();
-    }  
+      await this.useLVMToggle.click();
+    }
 
     async back() {
-        await this.backButton.click();
+      await this.backButton.click();
+    }
+
+    async actionAddHomeFs() {
+      await expect(this.swapPartitionText).toBeVisible({timeout: 2 * 60 * 1000});
```

Comment:
we don't need to check the current file systems, in the future we might have different default layout in a different product installable with Agama and we would need to update this validations.

Reply:
Playwright has many default check before clicking on something (among them the enable afir, you might just need to increase the timeout of the action below, could I see the failure?